compiler/test/dotty/tools/vulpix/ParallelTesting.scala

```diff
@@ -456,13 +461,20 @@ trait ParallelTesting extends RunnerOrchestration { self =>
       val toolArgs = toolArgsFor(files.toList.map(_.toPath), getCharsetFromEncodingOpt(flags0))
 
       val spec = raw"(\d+)(\+)?".r
-      val testFilter = toolArgs.get(ToolName.Test) match
+      val testIsFiltered = toolArgs.get(ToolName.Test) match
         case Some("-jvm" :: spec(n, more) :: Nil) =>
           if more == "+" then isJavaAtLeast(n) else javaSpecVersion == n
         case Some(args) => throw new IllegalStateException(args.mkString("unknown test option: ", ", ", ""))
         case None => true
+      val minRelease = toolArgs.get(ToolName.Test) match
+        case Some("-jvm" :: spec(n, more) :: Nil) =>
+          if more == "+" then javaSpecVersion else n
+        case _ => "8"
 
-      def scalacOptions = toolArgs.get(ToolName.Scalac).getOrElse(Nil)
+      def scalacOptions =
+        toolArgs.get(ToolName.Scalac)
+          .map(args => if args.exists(_.startsWith("-release")) then args else "-release" :: minRelease :: args)
```

Comment:
actually defaulting `-release 8` was deemed problematic, so at best it should be introduced under a test flag of some sort.